src/eml2/HdfProxy.h

```diff
@@ -362,6 +362,9 @@ namespace EML2_NS
 		/** The compression level used for writing data */
 		unsigned int compressionLevel = 0;
 
+		/** The dataset property list Id */
+		hdf5_hid_t dsetPlistId = 0; //properly should be H5P_DEFAULT
```

Comment:
I think I would prefer a std::unique_ptr on hdf5_hid_t intialized to empty.
I am afraid that H5P_DEFAULT could change its integer value in different HDF5 version.
It would then require using ternary operator where dsetPlistId is used.

Reply:
`std::unique_ptr<hdf5_hid_t> dsetPlistId;` ?
And then dsetPlistId.get() ? It will return nullptr.

Reply:
yes.
But get() is not necessary most of time I would say since [operator bool](https://en.cppreference.com/w/cpp/memory/unique_ptr/operator_bool) works on unique_ptr.
And [operator*](https://en.cppreference.com/w/cpp/memory/unique_ptr/operator*) as well.

Reply:
I don't understand your point. There are hid_t everywhere